db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -217,10 +222,14 @@ Status DBImpl::FlushMemTableToOutputFile(
       // CURRENT file. With current code, it's just difficult to tell. So just
       // be pessimistic and try write to a new MANIFEST.
       // TODO: distinguish between MANIFEST write and CURRENT renaming
-      auto err_reason = versions_->io_status().ok()
-                            ? BackgroundErrorReason::kFlush
-                            : BackgroundErrorReason::kManifestWrite;
-      error_handler_.SetBGError(io_s, err_reason);
+      if (!versions_->io_status().ok()) {
+        error_handler_.SetBGError(io_s, BackgroundErrorReason::kManifestWrite);
+      } else if (total_log_size_ > 0) {
+        error_handler_.SetBGError(io_s, BackgroundErrorReason::kFlush);
+      } else {
+        // If the WAL is empty, we use different error reason
+        error_handler_.SetBGError(io_s, BackgroundErrorReason::kFlushNoWAL);
```

Comment:
To be on the safe side, can we also check if ```SyncClosedLogs``` failed on line 184, and if so, set reason to ```kFlush```?

Reply:
Sure. Yeah, that part is missed.

Reply:
> To be on the safe side, can we also check if `SyncClosedLogs` failed on line 184, and if so, set reason to `kFlush`?

Sure. Yeah, that part is missed.